lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -1435,6 +1461,24 @@ internal class EvaluatingCompiler(
                     // bind the joined value to the bindings for the filter/project
                     FromProduction(joinedValues, fromEnv.nest(localsBinder.bindLocals(joinedValues)))
                 }
+            // Nest LET bindings in the FROM environment
+            if (compiledLetSources != null) {
+                seq = seq.map { fromProduction ->
+                    val parentEnv = fromProduction.env
+
+                    val letEnv: Environment = compiledLetSources.fold(parentEnv) { accEnvironment, curLetSource ->
```

Comment:
Similar question, is kotlin `fold` a *left* fold?

Reply:
In Kotlin, `fold` is a _left_ fold.

Per the [docs](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/fold.html):
"Accumulates value starting with initial value and applying operation from **left** to **right** to current accumulator value and each element."